docs/guides/postgres/backup/kubestash/overview/index.md

```diff
@@ -0,0 +1,98 @@
+
+5. KubeStash operator watches for `BackupConfiguration` objects.
+
+6. Once the KubeStash operator finds a `BackupConfiguration` object, it creates `Repository` with the information specified in the `BackupConfiguration`.
+
+7. KubeStash operator watches for `Repository` custom resources. When it finds the `Repository` object, it Initializes `Repository` by uploading `repository.yaml` file into the `spec.sessions[*].repositories[*].directory` path specified in `BackupConfiguration`.
+
+8. Then, it creates a `CronJob` for each session with the schedule specified in `BackupConfiguration` to trigger backup periodically.
+
+9. KubeStash operator triggers an instant backup as soon as the `BackupConfiguration` is ready. After that, backups are triggered by the `CronJob` according to the specified schedule.
+
+10. KubeStash operator watches for `BackupSession` custom resources.
+
+11. When it finds a `BackupSession` object, it creates a `Snapshot` custom resource for each `Repository` specified in the `BackupConfiguration`.
+
+12. Then it resolves the respective `Addon` and `Function` and prepares backup `Job` definition.
+
+13. Then, it creates the `Job` to backup the targeted `PostgreSQL` database.
+
+14. The backup `Job` reads necessary information (e.g. auth secret, port)  to connect with the database from the `AppBinding` crd. It also reads backend information and access credentials from BackupStorage crd, Storage Secret and Repository path respectively.
+
+15. Then, the `Job` dumps the targeted `PostgreSQL` database and uploads the output to the backend. KubeStash pipes the output of dump command to uploading process. Hence, backup `Job` does not require a large volume to hold the entire dump output.
+
+16. After the backup process is completed, the backup `Job` updates the `status.components[dump]` field of the `Snapshot` resources with backup information of the target `PostgreSQL` database.
+
+## How Restore Process Works
+
+The following diagram shows how KubeStash restores backed up data into a `PostgreSQL` database. Open the image in a new tab to see the enlarged version.
+
+<figure align="center">
+  <img alt="Database Restore Overview" src="/docs/guides/postgres/backup/kubestash/overview/images/restore_overview.svg">
+  <figcaption align="center">Fig: PostgreSQL Restore Process Overview</figcaption>
+</figure>
+
+The restore process consists of the following steps:
+
+1. At first, a user creates a `PostgreSQL` database where the data will be restored or the user can use the same `PostgreSQL` database.
+
+2. Then, she creates a `RestoreSession` custom resource that specifies the target `PostgreSQL` database where the backed-up data will be restored. the `Repository` object that points to a `BackupStorage` that holds backend information, and the target `Snapshot`, which will be restored. It also specifies the `Addon` info with task to use to restore.
```

Comment:
Then, she creates a RestoreSession custom resource that specifies the target database where the backed-up data will be restored, addon information (including restore tasks), the target snapshot to be restored, the Repository containing that snapshot, and other additional settings.